### ISSUE TITLE
Enable Search tests in sovereign cloud

### ISSUE THREAD

**Issue by xiangyan99** [Azure/azure-sdk-for-python#22867]
https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/604/Sovereign-Cloud-Testing for more information

**Comment by v-xuto** [comment]
@benbp Please help add `test-resources.json` file in search. It runs failed at section `Deploy test resources` on Usgov and China. For more details please check [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1351370&view=logs&j=30224f6e-6ea4-5543-09d9-1149d1110260&t=74c31dd1-0da8-5e2f-2601-0b387e287e93):

**Comment by xiangyan99** [comment]
We use the PS script with the json to provision resources, right?

We can use the same PS script upon bicep file. (I see you hard coded the location in bicep file so maybe we need to update it)

**Comment by v-xuto** [comment]
@xiangyan99 Yes, your understanding is correct. We may need to replace the bicep file with `test_resource.json`.

**Comment by kashifkhan** [comment]
@xiangyan99 I think you are right about updating the bicep hardcoded location to simply be `location`. Perhaps a ternary for now since only China East 2 supports AI Enrichment.

<img width="980" alt="image" src="https://user-images.githubusercontent.com/361477/153216131-9825490a-7c3c-45cb-bd2f-d6b2ed8cd5b5.png">


The only issue I foresee is around Semantic Search, since that is not supported at all.

**Comment by benbp** [comment]
@v-xuto the bicep file is equivalent to the ARM template (and compiles down to an ARM template at deploy time). I think @kashifkhan's above update looks like it will work.

**Comment by v-xuto** [comment]
@kashifkhan Thanks for updating the bicep file.But, you may forgot to add Usgov's searchEndpointSuffix, we are not familiar with bicep, so not sure how to change it. 

@benbp Thanks for the explanation, I got it. Because Usgov's searchEndpointSuffix is missing from the bicep file, and we are not familiar with it. So, we replaced bicep with the test_resource.json file. Besides, I removed the properties semanticSearch in the search service, and all three clouds can pass. I'm not sure, is it appropriate to do this?
And the Search PR https://github.com/Azure/azure-sdk-for-python/pull/23005 has been created, please review.
